rotary/rules/rotary.smk

```diff
@@ -312,16 +367,180 @@ rule qc_long_length_stats:
         length_stats.to_csv(output[0], sep='\t', header=None, index=True)
 
 
+rule finalize_qc_long:
+    input:
+        "{sample}/qc/long/{sample}_nanopore_qc.fastq.gz"
+    output:
+        "{sample}/qc/{sample}_QC_long.fastq.gz"
+    run:
+        source_relpath = os.path.relpath(str(input), os.path.dirname(str(output)))
+        os.symlink(source_relpath, str(output))
+
+
 rule qc_long:
     input:
-        expand("{sample}/stats/{sample}_qc_long_length_stats.txt",sample=SAMPLE_NAMES)
+        expand("{sample}/qc/{sample}_QC_long.fastq.gz", sample=SAMPLE_NAMES),
+        expand("{sample}/stats/{sample}_qc_long_length_stats.txt", sample=SAMPLE_NAMES)
     output:
         temp(touch("checkpoints/qc_long"))
 
 
+rule short_read_reformat:
+    """
+    Makes the input format of the short reads consistent (e.g., by trimming read descriptions and tossing invalid 
+    nucleotide characters). Threads are locked at a maximum of 4 because this code is IO limited.
```

Comment:
@jmtsuji I think snakemake has a way to override resources for specific rules. What is interesting about this is that if you put this code on a RAM disk or potentially SSD you can use more threads. I think locking it at 4 threads is good for now.

Reply:
I found that when I was running HMMER3 on ZFS data with the RAM cache I could use more threads than the 8 it normally bottlenecks at because the RAM cache was able to feed the algorithm fast enough to make a difference.

Reply:
Good point that some hardware configurations could support higher thread counts. One good thing is that snakemake should be able to run multiple jobs in parallel that are below the total resources allotted to snakemake. So even if capping this job at 4 threads does not maximize the thread usage in all setups, at least other rules should be able to run simultaneously to fill the space. I agree, let's keep it at 4 threads for now, but we can always change this in future (or add another config variable somewhere, whether exposed or hidden, for the thread count of small thread jobs).